test/rules/use_function_declaration_instead_variable.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N use_function_declaration_instead_variable`
+
+void main() {
+  thisIsGood() { // OK
+    print('this is good');
+  }
+
+  var bad1 = () { // LINT
+    print('this is bad');
+  };
+
+  var bad2 = bad1; // LINT
+
+  bad2 = bad1; // LINT
+
+  bad2 = () { // LINT
```

Comment:
This shouldn't be a lint for the same reason. You can't create lints for assignments because there is no alternative with the same semantics.